src/datajudge/constraints/base.py

```diff
@@ -121,12 +123,12 @@ def __init__(
         self,
         ref: DataReference,
         *,
-        ref2: Optional[DataReference] = None,
-        ref_value: Optional[Any] = None,
-        name: Optional[str] = None,
-        output_processors: Optional[
-            Union[OutputProcessor, List[OutputProcessor]]
-        ] = output_processor_limit,
+        ref2: DataReference | None = None,
+        ref_value: Any = None,
+        name: str | None = None,
+        output_processors: OutputProcessor
+        | list[OutputProcessor]
+        | None = output_processor_limit,
```

Comment:
Why a None did sneak in here?

Reply:
Before this change, `output_processors` annotated with `Optional[Union[OutputProcessor, List[OutputProcessor]]`. 

If we convert the `Optional`, this given us:
`Union[OutputProcessor, List[OutputProcessor] | None`.

If we now also covert the `Union` and (`List` to `list`), this gives us `OutputProcessor | list[OutputProcessor] | None`, which, afaict corresponds to the suggested change.

Does that make sense?

Reply:
Seems I missed a step in the chain of transformations.